api/apps/api/src/modules/scenarios/scenarios.service.ts

```diff
@@ -447,6 +482,50 @@ export class ScenariosService {
       },
     };
   }
+
+  private async mapCurrentPuStatusesAndClearRequested(
+    scenarioId: string,
+    kindToClear: string,
```

Comment:
I think you could type this more strictly as
```suggestion
    kindToClear: LockStatus,
```